index.js

```diff
@@ -713,6 +713,25 @@
       [$.Function, a, b, c, d],
       lift3);
 
+  //# on :: (b -> b -> c) -> (a -> b) -> a -> a -> c
+  //.
+  //. Takes a binary function, a unary function, and two values. Returns the
+  //. result of applying the unary function to each value, then passing the
+  //. results through the binary function.
+  //.
+  //. ```javascript
+  //. S.on(S.concat, S.reverse, [1, 2, 3], [4, 5, 6])
+  //. [3, 2, 1, 6, 5, 4]
+  //. ```
+  function on(f, g, x, y) {
+    return f(g(x), g(y));
+  }
+  S.on =
+  def('on',
+  {},
+  [$.Function, $.Function, a, a, c],
+  on);
+
   //. ### Composition
```

Comment:
Perhaps `on` should live in the __Composition__ section rather than the __Function__ section. When I consider what it does I imagine something like this:

       x           y
       |           |
       |           |
       |           |
       |           |
       |           |
    +-----+     +-----+
    |  g  |     |  g  |
    +-----+     +-----+
        \         /
         \       /
     g(x) \     / g(y)
           \   /
            \ /
          +-----+
          |  f  |
          +-----+
             |
             |
             |
             |
             |
       f(g(x))(g(y))